### ISSUE TITLE
Pull request, straight from GitHub desktop

### ISSUE THREAD

**Issue by mmgfrcs** [desktop/desktop#5015]
Well, as a small team who uses GitHub desktop in full, switching between the web and the client is a bit cumbersome. While it won't matter for large-scale projects (such as this one), it usually boils down to convenience. I would like it if we can post comments, issues and basically collaborate, straight from GitHub desktop.

As a first step, I would like it if we can create, manage and merge pull requests straight from GitHub desktop. What we have now basically just listing pull requests and commit against it. It's great, but not enough.

Creating pull requests follow the web convention, with changes:
- The branch must have changes (obviously). Disable this feature if the current branch has no committed changes.
- In the web, we choose which branch to pull from and to (source and destination branch). For the client implementation, I would suggest using the current branch as source, and give a dialog similar to the Merge branch dialog (this one) to choose the destination (usually master, but oh well). 

![image](https://user-images.githubusercontent.com/18462068/41808967-e8fe9e2c-770f-11e8-859d-7844a9b27a17.png)
*(Yay dark themes!)*

- The user press "Continue"-like button, then present them with the commit diff. Continuing from #4588, the client should tell users for conflicts that prevented a clean merge like this on the web:
![image](https://user-images.githubusercontent.com/18462068/41809031-1e4375b6-7711-11e8-9715-17506dc33b50.png). 

    There should be a dialog similar to the commit box (this) where users could type the name and description of the pull request, then click the button to post the request.

![image](https://user-images.githubusercontent.com/18462068/41809101-d07e2cb2-7711-11e8-8f25-672266c0ae40.png)

Basically, after creating the pull request, ~~GitHub desktop then turns into Discord where you can post comments and stuff.~~ well, not exactly. What's going to happen is that there will be a new pane on the right of the client where you can post comments and such, while keeping the same UI as normal pull request commit.

Of course, this comes with a consequence of getting _these_ in
![image](https://user-images.githubusercontent.com/18462068/41809198-8784be20-7713-11e8-9d02-2f03c42d5562.png)

**Comment by mmgfrcs** [comment]
...and things, right there, starts to go south

Damn Enter button just send the issue, and I haven't even finished typing yet. Gotta wait for others before I would finish this

This workflow could extend to issues too, where members could create issues and discuss it in the client.

**Comment by j-f1** [comment]
@mmgfrcs Although we probably won’t be adding this in the near future, you can currently hit Cmd (or Ctrl) + R to open the “create pull request” page in your browser for the current branch.

**Comment by mmgfrcs** [comment]
@j-f1 that I know, and that I use every time (Branch > Create pull request)
It's more to reduce unnecessary conflicts and speed up collaboration.

**Comment by shiftkey** [comment]
@mmgfrcs thanks for opening this issue!

We had a lengthy discussion about this feature (which was in the classic apps) in #295, and for the moment it's not on our roadmap because we struggled to differentiate ourselves with what is currently available on GitHub The Website.

I'm going to cross-reference this issue in #295, but please add anything else to that discussion.
